gitapp/lib/user_details/user_details_repository.dart

```diff
@@ -0,0 +1,39 @@
+import 'dart:convert';
+import 'package:gitapp/user_details/user_details_model.dart';
+import 'package:gitapp/user_details/user_repo_model.dart';
+import "package:http/http.dart" as http;
+
+class UserDetailsRepository {
+  UserDetailsRepository();
+  static final UserDetailsRepository userDetailsRepository =
+      new UserDetailsRepository();
+
+  Future<UserDetails> fetchUserDetails(int userId) async {
+    String url = "https://api.github.com/user/$userId";
+    final response = await http.get(Uri.parse(url));
+    if (response.statusCode == 200) {
+      return UserDetails.fromJSON(json.decode(response.body));
+    } else {
+      return UserDetails(
+          id: 0,
+          avatarUrl: '',
+          username: '',
+          followersCount: 0,
+          repositoriesCount: 0);
+    }
+  }
+
+  Future<List<UserRepo>> fetchRepositoriesList(int userId, currentPage) async {
+    List<UserRepo> listRepos = <UserRepo>[];
+    String url =
+        "https://api.github.com/user/$userId/repos?per_page=3&$currentPage";
+    final response = await http.get(Uri.parse(url));
+    if (response.statusCode == 200) {
```

Comment:
Do we need this if  and else ? either way you are returning the listRepo. The else could be removed and return could be moved outside of the if closure

Reply:
Done